Sources/Public/DotLottie/DotLottieFileHelpers.swift

```diff
@@ -289,4 +289,21 @@ extension DotLottieFile {
     }
   }
 
+  /// Loads an DotLottie from a data synchronously. Returns a `Result<DotLottieFile, Error>`
+  /// - Parameters:
+  ///   - data: The data(`Foundation.Data`) object to load DotLottie from
+  ///   - filename: The name of the lottie file without the lottie extension. eg. "StarAnimation"
+  public static func loadedFrom(
+    data: Data,
+    filename: String)
+    -> Result<DotLottieFile, Error>
+  {
+    do {
+      let dotLottieFile = try DotLottieFile(data: data, filename: filename)
```

Comment:
`DotLottieFile.init` performs a lot of work on the calling thread (it decompresses a zip file). This method should be asynchronous with a completion callback so the work isn't done on the main thread. We should also provide an `async` version that wraps the implementation that takes a completion handler (using `withCheckedThrowingContinuation`).

Reply:
Done.